File: Java/Desktop/ReadyToExam/src/NewYearGift/Model/Logic/Sort.java
package NewYearGift.Model.Logic;

import NewYearGift.Model.Entity.Abstract.Candy;
import NewYearGift.Model.Entity.Container.Gift;

public class Sort {

    public static void insert(Gift gift) {
        Candy temp;
        int j;
        for(int i = 0; i < gift.getSize() - 1; i++){
            if (gift.getCandy(i).getPrice() > gift.getCandy(i + 1).getPrice()) {
                temp = gift.getCandy(i + 1);
                gift.changeCandy(gift.getCandy(i),i + 1);
                j = i;
                while (j > 0 && temp.getPrice() < gift.getCandy(j - 1).getPrice()) {
                    gift.changeCandy(gift.getCandy(j - 1),j);
                    j--;
                }
                gift.changeCandy(temp,j);
            }
        }
    }



    public static void bubble(Gift gift) {
        boolean sorted = true;
        for (int i = gift.getSize(); i > 0; i--) {
            for (int j = 0; j < i - 1; j++) {
                if (gift.getCandy(j).getPrice() > gift.getCandy(j + 1).getPrice()) {
                    Candy tmp = gift.getCandy(j);
                    gift.changeCandy(gift.getCandy(j + 1), j);
                    gift.changeCandy(tmp, j + 1);
                    sorted = false;
                }
                if (sorted){
                    break;
                }

            }
        }
    }

}

File: Csharp/ReadyToExam/ReadyToExam/NewYearGift/Model/Logic/Sort.cs
using ReadyToExam.NewYearGift.Model.Abstracts;
using ReadyToExam.NewYearGift.Model.Entity.Container;

namespace ReadyToExam.NewYearGift.Model.Logic
{
    public class Sort
    {
        public static void Insert(Gift gift) {
            Candy temp;
            int j;
            for(int i = 0; i < gift.GetSize() - 1; i++){
                if (gift.GetCandy(i).Price > gift.GetCandy(i + 1).Price) {
                    temp = gift.GetCandy(i + 1);
                    gift.ChangeCandy(gift.GetCandy(i),i + 1);
                    j = i;
                    while (j > 0 && temp.Price < gift.GetCandy(j - 1).Price) {
                        gift.ChangeCandy(gift.GetCandy(j - 1),j);
                        j--;
                    }
                    gift.ChangeCandy(temp,j);
                }
            }
        }

        public static void Bubble(Gift gift) {
            bool sorted = true;
            for (int i = gift.GetSize(); i > 0; i--) {
                for (int j = 0; j < i - 1; j++) {
                    if (gift.GetCandy(j).Price > gift.GetCandy(j + 1).Price) {
                        Candy tmp = gift.GetCandy(j);
                        gift.ChangeCandy(gift.GetCandy(j + 1), j);
                        gift.ChangeCandy(tmp, j + 1);
                        sorted = false;
                    }
                    if (sorted){
                        break;
                    }
                }
            }
        }        
    }
}